var/spack/repos/builtin/packages/openscenegraph/package.py

```diff
@@ -32,13 +36,17 @@ def cmake_args(self):
         args = [
             '-DDYNAMIC_OPENSCENEGRAPH={0}'.format(shared_status),
             '-DDYNAMIC_OPENTHREADS={0}'.format(shared_status),
-            '-DZLIB_INCLUDE_DIR={0}'.format(spec['zlib'].prefix.include),
-            '-DZLIB_LIBRARY={0}/libz.{1}'.format(spec['zlib'].prefix.lib,
-                                                 dso_suffix),
             '-DBUILD_OSG_APPLICATIONS=OFF',
             '-DOSG_NOTIFY_DISABLED=ON',
             '-DLIB_POSTFIX=',
+            '-DCMAKE_RELWITHDEBINFO_POSTFIX=',
         ]
+        if spec.satisfies('@:3.2'):
+            args.extend([
+                '-DZLIB_INCLUDE_DIR={0}'.format(spec['zlib'].prefix.include),
+                '-DZLIB_LIBRARY={0}/libz.{1}'.format(spec['zlib'].prefix.lib,
+                                                     dso_suffix),
```

Comment:
But it still links to zlib in 3.2+ correct?

Reply:
Yes, older versions used a built-in method to find zlib, and setting the variables was required. Newer versions use cmake's module to find zlib and setting the older variables would break the build.